OrbitBase/include/OrbitBase/Profiling.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_PROFILING_H_
+#define ORBIT_BASE_PROFILING_H_
+
+#include "absl/time/time.h"
+
+#ifdef _WIN32
+#include <Windows.h>
+using pid_t = uint32_t;
+#else
+#include <sys/syscall.h>
+#include <unistd.h>
+#endif
+
+[[nodiscard]] inline uint64_t MonotonicTimestampNs() {
+  timespec ts;
+#ifdef _WIN32
+  __int64 time;
+  GetSystemTimeAsFileTime((FILETIME*)&time);
+  ts.tv_sec = time / 10000000i64;
+  ts.tv_nsec = time % 10000000i64 * 100;
```

Comment:
return static_cast<uint64_t>(time)*100?

Reply:
maybe use std::steady_clock::now() for windows and then get nanoseconds?

Reply:
Yes, much cleaner, thanks.  For steady_clock::now, what advantage does it offer? Better precision? Are we sure that it's returning the values in the same time domain as the ETW events? I'm not against it, but it requires an investigation which is really outside the scope of this PR, which, again, is shuffling existing code around... :-)

Reply:
just that it does not not use windows-specific api.